layabout.py

```diff
@@ -1,3 +1,325 @@
+        """
+        Format an error message for missing positional arguments.
+
+        Args:
+            name: The function name.
+            sig: The function's signature.
+            num_params: The number of function parameters.
+
+        Returns:
+            str: A formatted error message.
+        """
+        if num_params == 0:
+            plural = 's'
+            missing = 2
+            arguments = "'slack' and 'event'"
+        else:
+            plural = ''
+            missing = 1
+            arguments = "'event'"
+
+        return (f"{name}{sig} missing {missing} required positional "
+                f"argument{plural}: {arguments}")
+
+    def handle(self, type: str, *, kwargs: dict = None) -> Callable:
+        """
+        Register an event handler with the :obj:`Layabout` instance.
+
+        Args:
+            type: The name of a Slack RTM API event to be handled. As a
+                special case, although it is not a proper RTM event, ``*`` may
+                be provided to handle all events. For more information about
+                available events see the
+                `Slack RTM API <https://api.slack.com/rtm>`_.
+            kwargs: Optional arbitrary keyword arguments passed to the event
+                handler when the event is triggered.
+
+        Returns:
+            A decorator that validates and registers a Layabout event handler.
+
+        Raises:
+            TypeError: If the decorated :obj:`Callable`'s signature does not
+                accept at least 2 parameters.
+        """
+        def decorator(fn: Callable) -> Callable:
+            # Validate that the wrapped callable is a suitable event handler.
+            sig = signature(fn)
+            num_params = len(sig.parameters)
+            if num_params < 2:
+                raise TypeError(self._format_parameter_error_message(
+                    fn.__name__, sig, num_params))
+
+            # Register a tuple of the callable and its kwargs, if any.
+            self._handlers[type].append((fn, kwargs or {}))
+
+            @wraps(fn)
+            def wrapper(slack: SlackClient, event: Dict[str, Any],
+                        **kwargs: Any) -> Any:
+                # We don't actually do anything with the return value, but this
+                # might make unit testing easier for users.
+                return fn(slack, event, **kwargs)
+            return wrapper
+        return decorator
+
+    def _ensure_slack(self, connector: Any, retries: int,
+                      backoff: Callable[[int], float]) -> None:
+        """ Ensure we have a SlackClient. """
+        connector = self._env_var if connector is None else connector
+        slack: SlackClient = _create_slack(connector)
+        self._slack = _SlackClientWrapper(
+            slack=slack,
+            retries=retries,
+            backoff=backoff
+        )
+
+    def run(self, *, connector: Union[EnvVar, Token, SlackClient, None] = None,
```

Comment:
I think `connector` should be on the next line as well. that way all arguments are aligned on the same column and so you can vertically scan the keyword argument names